File: src/slm_training/features/keys.py
"""Canonical OpenFeature flag keys for product experiments."""

from __future__ import annotations

from typing import Final

DASHBOARD_DEFAULT_RENDERER: Final = "dashboard.default-renderer"
VSS_DECODE_ENABLED: Final = "vss.decode-enabled"
PLAYGROUND_GRAMMAR_CONSTRAINED_DEFAULT: Final = "playground.grammar-constrained-default"

PRODUCT_FLAG_KEYS: Final[tuple[str, ...]] = (
    DASHBOARD_DEFAULT_RENDERER,
    VSS_DECODE_ENABLED,
    PLAYGROUND_GRAMMAR_CONSTRAINED_DEFAULT,
)

File: src/apps/dashboard/src/features/keys.ts
export const DASHBOARD_DEFAULT_RENDERER = "dashboard.default-renderer";
export const VSS_DECODE_ENABLED = "vss.decode-enabled";
export const PLAYGROUND_GRAMMAR_CONSTRAINED_DEFAULT =
  "playground.grammar-constrained-default";

export const PRODUCT_FLAG_KEYS = [
  DASHBOARD_DEFAULT_RENDERER,
  VSS_DECODE_ENABLED,
  PLAYGROUND_GRAMMAR_CONSTRAINED_DEFAULT,
] as const;

export type ProductFlagKey = (typeof PRODUCT_FLAG_KEYS)[number];

export type DashboardRenderer = "compiled" | "interpreted";
